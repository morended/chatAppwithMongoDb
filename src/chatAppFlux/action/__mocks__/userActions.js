var userActionsMock = jest.genMockFromModule('./../userActions.js');

module.exports = userActionsMock;
