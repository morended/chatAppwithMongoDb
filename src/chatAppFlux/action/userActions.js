import alt from "../alt.js"
import http from "superagent"

class UserActions {

    registerUser(payload) {
        return (dispatch) => {
            http
                .post('/api/users/register')
                .set('Content-Type', 'application/json')
                .query(payload)
                .send(payload)
                .end((err, res) => this.handleResponse(err, res, dispatch));
        }
    }

    loginUser(payload){
        return (dispatch) => {
            http
                .post('/api/users/login')
                .set('Content-Type', 'application/json')
                .query(payload)
                .send(payload)
                .end((err, res) => this.handleResponse(err, res, dispatch));
        }
    }

    handleResponse(error, result, dispatch) {
        console.log('in handle response');
        console.log(result);
        if (error) {
            console.log('An error occurred:', error);
            alert('An error occurred: ', error);
            return;
        }
        dispatch({
            data: result.text
        });
    }
}

export default alt.createActions(UserActions);
