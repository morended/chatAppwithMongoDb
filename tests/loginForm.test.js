import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

import userActions from '../src/chatAppFlux/action/userActions.js';
import LoginForm from '../src/chatAppFlux/components/loginForm.js';

jest.mock('../src/chatAppFlux/action/userActions.js');

describe('Login form component (snapshot)', () => {

  it('renders correctly', () => {
    const component = renderer.create(
      <LoginForm/>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});

describe('LoginForm component', () => {
  let component;
  beforeEach(() => {
    component = shallow(
      <LoginForm/>
    );
  });

  it('renders the title', () => {
    expect(component.find('h1').text()).toContain("Login to your account");
  });

  it('renders two text boxes', () => {
    expect(component.find('input').length).toEqual(2);
  });

  it('sets initial state', () => {
    expect(component.state().username).toEqual('');
    expect(component.state().password).toEqual('');
  });

  it('contains 2 divs with class name "row"', () => {
    expect(component.find('div.row').length).toEqual(2);
  });

  it('properly wraps username input in tags', () => {
    expect(component.find('div.row').first().children().length).toEqual(1);
    expect(component.find('.form-group').first().children().length).toEqual(2);
    expect(component.find('.control-label').first().text()).toContain('Username');
  });

  it('properly wraps password input in tags', () => {
    expect(component.find('div.row').at(1).children().length).toEqual(1);
    expect(component.find('.form-group').at(1).children().length).toEqual(2);
    expect(component.find('.control-label').at(1).text()).toContain('Password');
  });

  it('renders Login button', () => {
    let button = component.find('button');
    expect(button.length).toEqual(1);
    expect(button.text()).toEqual('Login');
  });

  it('updates state on username change', () => {
    let evnt = { target : {name: 'username', value: 'test'} };
    component.find('input').first().simulate('change', evnt);
    expect(component.state().username).toEqual(evnt.target.value);
  });

  it('updates state on password change', () => {
    let evnt = { target : {name: 'password', value: 'pass'} };
    component.find('input').at(1).simulate('change', evnt);
    expect(component.state().password).toEqual(evnt.target.value);
  });

  it('resets state on button click', () => {
    let evnt = { target : { value: 'test'} };
    evnt.preventDefault = jest.fn();
    component.find('button').simulate('click', evnt);
    expect(component.state().username).toEqual('');
    expect(component.state().password).toEqual('');
  });
});
