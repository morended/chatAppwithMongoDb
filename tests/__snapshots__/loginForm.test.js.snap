// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form component (snapshot) renders correctly 1`] = `
<form
  onSubmit={[Function]}
  role="form"
>
  <h1>
    Login to your account
  </h1>
  <div
    className="row"
  >
    <div
      className="form-group col-sm-4"
    >
      <label
        className="control-label"
      >
        Username
      </label>
      <input
        className="form-control"
        name="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="form-group col-sm-4"
    >
      <label
        className="control-label"
      >
        Password
      </label>
      <input
        className="form-control"
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-primary btn-lg"
    >
      Login
    </button>
  </div>
</form>
`;
